tests/test_writer.py

```diff
@@ -64,8 +70,58 @@ def test_format_requirement_environment_marker(from_line, writer):
     ireq = from_line('test ; python_version == "2.7" and platform_python_implementation == "CPython"')
     reverse_dependencies = set()
 
-    result = writer._format_requirement(
+    result = writer()._format_requirement(
         ireq, reverse_dependencies, primary_packages=['test'],
         marker=ireq.markers)
     assert (result ==
             'test ; python_version == "2.7" and platform_python_implementation == "CPython"')
+
+
+def test_iter_lines__do_not_allow_unsafe_packages(mocker, from_line, writer):
+    unsafe_package = UNSAFE_PACKAGES.copy().pop()
+    unsafe_dep = '{}==1.2'.format(unsafe_package)
+    ireq = from_line(unsafe_dep)
+    w = writer(allow_unsafe=False)
+    mocker.patch.object(w, 'write_header', return_value=['header'])
+    mocker.patch.object(w, 'write_flags', return_value=['flags'])
+    mocker.patch.object(w, '_sort_key', return_value=(ireq.editable, str(ireq.req).lower()))
+    format_requirement = mocker.patch.object(
+        w, '_format_requirement', return_value='{}'.format(unsafe_package)
+    )
+    comment = mocker.patch(
+        'piptools.writer.comment', side_effect=['foobar', '# {}'.format(unsafe_package)])
```

Comment:
Unit tests should test functionality in isolation of other modules assuming that those modules are tested independently.